### ISSUE TITLE
Mesosphere maven repos throw a 403

### ISSUE THREAD

**Issue by aaronjwood** [mesosphere/dcos-cassandra-service#253]
http://downloads.mesosphere.com/maven-snapshot
http://downloads.mesosphere.com/maven

These URLs both give a 403. I'm unable to grab all of the dependencies needed with Gradle for the project because of this. Compilation fails when running `./gradlew clean build`:

```
:cassandra-commons:compileJava
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:2291: error: cannot find symbol
      com.google.protobuf.GeneratedMessageV3 implements
                         ^
  symbol:   class GeneratedMessageV3
  location: package com.google.protobuf
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:848: error: cannot find symbol
      com.google.protobuf.GeneratedMessageV3 implements
                         ^
  symbol:   class GeneratedMessageV3
  location: package com.google.protobuf
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:1489: error: cannot find symbol
      com.google.protobuf.GeneratedMessageV3 implements
                         ^
  symbol:   class GeneratedMessageV3
  location: package com.google.protobuf
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:6904: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    com.google.protobuf.GeneratedMessageV3.FieldAccessorTable
                                          ^
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:6909: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    com.google.protobuf.GeneratedMessageV3.FieldAccessorTable
                                          ^
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:6914: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    com.google.protobuf.GeneratedMessageV3.FieldAccessorTable
                                          ^
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:6919: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    com.google.protobuf.GeneratedMessageV3.FieldAccessorTable
                                          ^
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:6924: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    com.google.protobuf.GeneratedMessageV3.FieldAccessorTable
                                          ^
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:2295: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    private CassandraConfig(com.google.protobuf.GeneratedMessageV3.Builder<?> builder) {
                                                                  ^
/Users/user/Code/src/dcos-cassandra-service/cassandra-commons/src/generated/main/java/com/mesosphere/dcos/cassandra/common/CassandraProtos.java:2435: error: package com.google.protobuf.GeneratedMessageV3 does not exist
    protected com.google.protobuf.GeneratedMessageV3.FieldAccessorTable
```
etc. etc. etc.

It looks like this has happened before as well https://github.com/mesosphere/chaos/issues/23

**Comment by aaronjwood** [comment]
Looks like adding a trailing slash fixes the issue:
http://downloads.mesosphere.com/maven-snapshot/
http://downloads.mesosphere.com/maven/

**Comment by aaronjwood** [comment]
There seems to be an issue with IntelliJ handling these URLs since they do have trailing slashes in `build.gradle`:

<img width="686" alt="screen shot 2016-09-20 at 4 02 51 pm" src="https://cloud.githubusercontent.com/assets/446283/18686940/db621c34-7f4b-11e6-9fdd-6c4e228f628d.png">

**Comment by aaronjwood** [comment]
Okay so once I cleaned out `cassandra-commons/src/generated`, uninstalled the latest version of protobuf (`brew uninstall protobuf`), installed protobuf 2.5.0 (`brew install homebrew/versions/protobuf250`), and tried to build again everything worked.

I was thinking that the project would bring down the required libs, not that I needed a system-wide install of an older version of protobuf.

I think it'd be nice to have this provided by the project so that people don't need to downgrade protobuf on their system and possibly break other projects that rely on newer versions. I believe we could bring down the right version of `protoc` using Gradle.

**Comment by aaronjwood** [comment]
If `protoc` 2.5.0 can be published to Maven we should be able to contain everything within the project. Then we can just have Gradle bring down the artifact and not rely on people mucking with their system. 

I opened an issue about this here https://github.com/google/protobuf/issues/2147

**Comment by bzamecnik** [comment]
It seems the proto files are compiled with 3.x compiler, while linked to 2.x JAR.

On brew/OSX you can have latest protobuf installed along with protobuf@2.5. The 2.5 binary is available in ` /usr/local/opt/protobuf@2.5/bin/protoc`.
